portality/models/v2/journal.py

```diff
@@ -817,6 +819,20 @@ def propagate_in_doaj_status_to_articles(self):
             article.set_in_doaj(self.is_in_doaj())
             article.save()
 
+    def reject_update_requests(self):
+        if self.current_application:
+            from portality.models.v2.application import Application
+            ca = Application.pull(self.current_application)
+            if ca is None:
+                raise exceptions.IllegalStatusException(message="Data mismatch: current application of this journal does not exist")
+            ca.add_note("Update request was automatically rejected because the associated journal was withdrawn or "
+                        "deleted.")
+            ca.save()
+            from portality.bll.doaj import DOAJ
+            svc = DOAJ.applicationService()
+            svc.reject_application(ca, current_user._get_current_object())
```

Comment:
I don't love the positioning of this, there's a lot of application level mixing going on.  The models should not be calling services, I don't think.

Instead there should be a service who is responsible for rejecting update requests

Reply:
See my comment elsewhere about where to put this functionality.